acceptable-albatrosses/albatrosses_hub/authentication/tests.py

```diff
@@ -0,0 +1,152 @@
+from django.contrib.auth.models import User, AnonymousUser
+from django.test import TestCase, Client, RequestFactory
+from django.urls import reverse, resolve
+from .views import register_page, login_page, logout_page
+
+
+class AuthTestCase(TestCase):
+    """Class to perform unit tests of auth app."""
+
+    def test_register_url(self):
+        response = Client().get(reverse("authentication:register"))
+        self.assertEqual(response.status_code, 200)
+
+    def test_login_url(self):
+        response = Client().get(reverse("authentication:login"))
+        self.assertEqual(response.status_code, 200)
+
+    def test_logout_url(self):
+        response = Client().get(reverse("authentication:logout"))
+        self.assertIn(response.status_code, (200, 302))
+
+    def test_register_use_template(self):
+        response = Client().get(reverse("authentication:register"))
+        self.assertTemplateUsed(response, "register.html")
+
+    def test_login_use_template(self):
+        response = Client().get(reverse("authentication:login"))
+        self.assertTemplateUsed(response, "login.html")
+
+    def test_logout_not_use_template(self):
+        response = Client().get(reverse("authentication:logout"))
+        self.assertTemplateNotUsed(response, "logout.html")
+
+    def test_register_using_register_func(self):
+        found = resolve(reverse("authentication:register"))
+        self.assertEqual(found.func, register_page)
+
+    def test_login_using_login_func(self):
+        found = resolve(reverse("authentication:login"))
+        self.assertEqual(found.func, login_page)
+
+    def test_logout_using_logout_func(self):
+        found = resolve(reverse("authentication:logout"))
+        self.assertEqual(found.func, logout_page)
+
+
+class AuthViewTest(TestCase):
```

Comment:
This `TestCase` seems to do a number of tests related to logging in and out of the website. Django provides some tools that you could consider to help you test that process, like [`django.test.Client.login`](https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.Client.login) and the methods listed below that.

While you seem to be testing the main authentication flow here, you could use those login helpers to test what happens when a logged in user versus an anonymous user does something without forcing an interdependence between your typical login request flow and the action that follows after it in your tests. (In other words, by using these helpers, you could test whether or not a logged in user can access "resource x" even if the login flow itself is currently broken.)